packages/swapper/src/swappers/thorchain/utils/getLimit/getLimit.ts

```diff
@@ -34,19 +54,56 @@ export const getLimit = async ({
   const isValidSlippageRange =
     bnOrZero(slippageTolerance).gte(0) && bnOrZero(slippageTolerance).lte(1)
   if (bnOrZero(expectedBuyAmountCryptoPrecision8).lt(0) || !isValidSlippageRange)
-    throw new SwapError('[getThorTxInfo]: bad expected buy amount or bad slippage tolerance', {
+    throw new SwapError('[getLimit]: bad expected buy amount or bad slippage tolerance', {
       code: SwapErrorTypes.BUILD_TRADE_FAILED,
       details: { expectedBuyAmountCryptoPrecision8, slippageTolerance },
     })
 
-  const tradeFeeCryptoPrecision8 = toBaseUnit(
+  const buyAssetTradeFeeCryptoPrecision8 = toBaseUnit(
     bnOrZero(buyAssetTradeFeeUsd).div(buyAssetUsdRate),
     THORCHAIN_FIXED_PRECISION,
   )
 
+  const sellAssetPoolId = adapters.assetIdToPoolAssetId({ assetId: sellAsset.assetId })
+  const sellAssetChainSymbol = sellAssetPoolId?.slice(0, sellAssetPoolId.indexOf('.'))
+  const sellAssetAddressData = await getInboundAddressDataForChain(
+    deps.daemonUrl,
+    sellAssetChainSymbol,
+  )
+
+  // We want this in the buy asset crypto precision
+  const refundFeeBuyAssetCryptoPrecision8 = (() => {
```

Comment:
This is the crux of the logic with respect to factoring in the refund fee when setting the limit in the TX memo.